walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,126 @@
+      @Override
+      public int compare(Event a, Event b) {
+        return Long.compare(a.getWhen().start(), b.getWhen().start());
+      }
+    });
+
+    // Add earliest VALID event to updatedEvents first, to use as a comparison for subsequent event.
+    int firstEventNum = 0;
+    boolean findingFirstEvent = true;
+    Collection<String> meetingAttendees = request.getAttendees();
+    while (findingFirstEvent && firstEventNum < events.size()) {
+      Event event = eventsList.get(firstEventNum);
+      Collection<String> eventAttendees = event.getAttendees();
+
+      // If every event attendees is NOT a meeting attendee, ignore the event.
+      if (!meetingAttendees.containsAll(eventAttendees)) {
+        firstEventNum++;
+        continue;
+      }
+
+      firstEventNum++;
+      updatedEvents.add(event);
+      findingFirstEvent = false;
+    }
+
+    // Combine events if they overlap or contain each other so that we don't need to double count overlapping events.
+    int updatedEventsIdx = 0;
+    for (int i = firstEventNum; i < events.size(); i++) {
+      Event event = eventsList.get(i);
+      Collection<String> eventAttendees = event.getAttendees();
+
+      // If every event attendees is NOT a meeting attendee, ignore the event.
+      if (!meetingAttendees.containsAll(eventAttendees)) {
+        continue;
+      }
+
+      // Check for overlaps.
+      Event prevEvent = updatedEvents.get(updatedEventsIdx);
+      if (event.getWhen().overlaps(prevEvent.getWhen())) { // If overlaps, replace the current prevEvent, take the later ending time.
+        int laterTiming = 0;
+        if (prevEvent.getWhen().end() > event.getWhen().end()) {
+          laterTiming = prevEvent.getWhen().end();
+        } else {
+          laterTiming = event.getWhen().end();
+        }
+        
+        TimeRange newTimeRange = TimeRange.fromStartEnd(prevEvent.getWhen().start(), laterTiming, false);
+        updatedEvents.set(updatedEventsIdx, new Event("", newTimeRange, Collections.emptySet()));
+      } else {
+        updatedEvents.add(event);
+        updatedEventsIdx++;
+      }      
+    }
+
+    return updatedEvents;
+  }
+
+  private Collection<TimeRange> findSlotsAvailable(List<Event> updatedEvents, MeetingRequest request) {
+    Collection<TimeRange> slotsAvailable = new ArrayList<>();
+    int eventIdx = 0;
+    Event event = updatedEvents.get(eventIdx);
+    int currStartingTime = TimeRange.START_OF_DAY;
+    int currEndingTime = event.getWhen().start();
+
+    // Add timerange to available slots if the duration of the timerange is longer than the meeting duration.
+    while (currStartingTime < TimeRange.END_OF_DAY && currEndingTime <= TimeRange.END_OF_DAY) {
+      int dur = currEndingTime - currStartingTime;
+      if (dur >= (int) request.getDuration()) {
+        boolean inclusive = false;
+        if (currEndingTime == TimeRange.END_OF_DAY) {
+          inclusive = true;
+        }
```

Comment:
Nit:
```
boolean inclusive = currEndingTime == TimeRange.END_OF_DAY;
```

Also, since the condition simple, this intermediate variable is not needed (can be directly used).